pkcs11/pkcs11.go

```diff
@@ -1855,3 +1856,20 @@ func makeCParamRSAOAEP(hash *crypto.Hash) (C.CK_RSA_PKCS_OAEP_PARAMS_PTR, error)
 
 	return cParam, nil
 }
+
+
+func Decrypt(key crypto.PrivateKey, encryptedData []byte) ([]byte, error) {
+	rsaPriv, ok := key.(*rsaPrivateKey) 
+	if !ok {
+		return nil, fmt.Errorf("Decrypt Error: expected type rsaPrivateKey, got %T", key)
+	}
+	return rsaPriv.decryptRSA(encryptedData)
+}
+
+func Encrypt(key crypto.PrivateKey, data []byte) ([]byte, error) {
+	rsaPriv, ok := key.(*rsaPrivateKey) 
+	if !ok {
+		return nil, fmt.Errorf("Encrypt Error: expected type rsaPrivateKey, got %T", key)
+	}
+	return rsaPriv.encryptRSA(data)
```

Comment:
This code would be generic if there was an `Encryptor` interface (Or something similar).

As a starting point:

```golang
type interface PKCS11Encryptor {
   func Encrypt(plaintext []byte) ([]byte, err)
}
```